File: starterpacks/python/application.py
#!/usr/bin/env python

import asyncio
import os
import websockets

from bot import Bot
from bot_message import BotMessage, MessageType, Move
from game_message import GameMessage


async def run():
    uri = "ws://127.0.0.1:8765"

    async with websockets.connect(uri) as websocket:
        bot = Bot()
        if "TOKEN" in os.environ:
            await websocket.send(BotMessage(type=MessageType.REGISTER, token=os.environ["TOKEN"]).to_json())
        else:
            await websocket.send(BotMessage(type=MessageType.REGISTER, name="MyBot").to_json())

        await game_loop(websocket=websocket, bot=bot)


async def game_loop(websocket: websockets.WebSocketServerProtocol, bot: Bot):
    while True:
        try:
            message = await websocket.recv()
        except websockets.exceptions.ConnectionClosed:
            # Connection is closed, the game is probably over
            break
        game_message: GameMessage = GameMessage.from_json(message)

        print(f"\nTurn {game_message.game.tick}")

        next_move: Move = bot.get_next_move(game_message)
        await websocket.send(BotMessage(type=MessageType.MOVE, action=next_move, tick=game_message.game.tick).to_json())


if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(run())

File: starterpacks/csharp/Application.cs
using System;
using System.Net.WebSockets;
using System.Threading;
using System.Threading.Tasks;
using System.Text;
using Newtonsoft.Json;
using Blitz2020;
using Newtonsoft.Json.Serialization;

public static class Application
{
    public static void Main(string[] args)
    {
        Task t = startClient();
        t.Wait();
    }


    public async static Task startClient(string address = "127.0.0.1:8765")
    {
        using (ClientWebSocket webSocket = new ClientWebSocket())
        {
            Uri serverUri = new Uri("ws://" + address);
            Bot bot = new Bot();

            await webSocket.ConnectAsync(serverUri, CancellationToken.None);

            string? token = Environment.GetEnvironmentVariable("TOKEN");
            string registerPayload = "";

            if (token == null)
            {
                registerPayload = "{\"type\": \"register\", \"name\": \"" + Bot.NAME + "\"}";
            }
            else
            {
                registerPayload = "{\"type\": \"register\", \"token\": \"" + token + "\"}";
            }

            await webSocket.SendAsync(
                new ArraySegment<byte>(Encoding.UTF8.GetBytes(registerPayload)),
                WebSocketMessageType.Text, true, CancellationToken.None);

            while (webSocket.State == WebSocketState.Open)
            {
                DefaultContractResolver contractResolver = new DefaultContractResolver
                {
                    NamingStrategy = new SnakeCaseNamingStrategy()
                };

                GameMessage message = JsonConvert.DeserializeObject<GameMessage>(await readMessage(webSocket), new JsonSerializerSettings
                {
                    ContractResolver = contractResolver,
                    Formatting = Formatting.Indented
                });

                if (message != null)
                {
                    Player.Move move = bot.nextMove(message);

                    await webSocket.SendAsync(
                        Encoding.UTF8.GetBytes("{\"type\": \"move\", \"action\": \"" + move.ToString() + "\", \"tick\": " + message.game.tick + "}"),
                        WebSocketMessageType.Text,
                        true, CancellationToken.None);
                }
            }
        }
    }

    public async static Task<string> readMessage(ClientWebSocket client)
    {
        string result = "";

        WebSocketReceiveResult receiveResult;
        do
        {
            ArraySegment<byte> segment = new ArraySegment<byte>(new byte[1024]);
            receiveResult = await client.ReceiveAsync(segment, CancellationToken.None);
            result += Encoding.UTF8.GetString(segment.Array);
        } while (!receiveResult.EndOfMessage);


        return result;
    }
}